### ISSUE TITLE
The complete SLAM pipeline cannot be built

### ISSUE THREAD

**Issue by 18709817706** [htkseason/VOLDOR#23]
run python setup_linux_full.py build_ext -i,There is an error.
../../gpu-kernels/meanshift.cu:14:94: error: default template arguments may not be used in function templates without -std=c++11 or -std=gnu++11
 at kernel_var, float* d_o_weight, float* d_o_weighted_space, int N, int dims) {
Would you please help me.

**Comment by htkseason** [comment]
Hello,
The error message seems related with c++ version you are using. What is your gcc/g++ version? Could you try follow the error message and add `-std=c++11` to the compiling?

**Comment by 18709817706** [comment]
Thank you for your answer. My question is cuda does not support it, but there is one question. Please help me.

Full pyvoldor module loaded.
Namespace(abs_resize=0.25, bf=192.19074, cx=304.77965, cy=86.427, disp_dir='/home/ymy/project/VOLDOR/KITTI-03/disp_mfn', enable_loop_closure='./ORBvoc.bin', enable_mapping=True, flow_dir='/home/ymy/project/VOLDOR/KITTI-03/flow_mfn', fx=360.76885, fy=360.76885, img_dir='/home/ymy/project/VOLDOR/KITTI-03/img', mode='stereo', resize=0.5, save_depths=None, save_poses=None)
Camera parameters set to 180.384425, 180.384425, 152.389825, 43.2135, 96.09537
Error: Loop closure not available. Cannot load pyDBoW3 module.

**Comment by 18709817706** [comment]
I have installed pyDBoW3, but it cannot be loaded.

**Comment by htkseason** [comment]
Have you moved the binary file pyDBoW3.pyd to your workspace folder?

**Comment by 18709817706** [comment]
Yes, I have put pyDBoW3. so in my workspace, but pyDBoW3 cannot be loaded.

**Comment by htkseason** [comment]
Can you remove the try catch segment in enable_loop_closure function, and let python print out the exact error message?

**Comment by 18709817706** [comment]
I put demo After removing the enable loop closure function under py, no error occurs, but pyDBoW3 cannot be loaded.

**Comment by 18709817706** [comment]
After I removed the enable loop closure function in demo.py, there was no error, but pyDBoW3 could not be loaded.

**Comment by htkseason** [comment]
Great to hear you having the system without loop closure running now. 
Could you try to remove the try..except surround `import pyDBoW3 as bow` in the `enable_loop_closure` function in `voldor_slam.py`? Doing this will allow us to see the error message so we can debug from there.

**Comment by wangshuo98** [comment]
hello，I have the same error
 I remove the try catch segment in enable_loop_closure function

ImportError: /home/VOLDOR/demo/pyDBoW3.so: undefined symbol: _ZN2cv3MatC2Ev

Looking forward to your response

**Comment by htkseason** [comment]
> hello，I have the same error I remove the try catch segment in enable_loop_closure function
> 
> ImportError: /home/VOLDOR/demo/pyDBoW3.so: undefined symbol: _ZN2cv3MatC2Ev
> 
> Looking forward to your response

Thanks for reporting the problem. Seems building the full-pipeline has some common compatibility issue. I never met this under my clean ubuntu 18 environment.  I will work on this when I have time. Meanwhile, you can first use VO-only, the performance (RPE) is actually very close.
